subsys/net/ip/net_if.c

```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
 	return NULL;
 }
 
+int net_if_get_name(struct net_if *iface, char *buf, int len)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL || len <= 0) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(net_if_get_config(iface)->name);
+	if (name_len > len) {
+		return -ERANGE;
+	}
+
+	strncpy(buf, net_if_get_config(iface)->name, name_len);
```

Comment:
Should this use `len` instead (the actual length of the provided buffer)?

Reply:
It should not matter here as we have made sure in the check few lines before, that `name_len` fits into `len` size buffer.